sdk/javascript/test/utils/arrayHelpers_spec.js

```diff
@@ -0,0 +1,268 @@
+const arrayHelpers = require('../../src/utils/arrayHelpers');
+const { expect } = require('chai');
+
+describe('arrayHelpers', () => {
+	describe('alignUp', () => {
+		const assertAlignUp = (range, alignment, expectedValue) => {
+			// Assert:
+			for (let i = range[0]; i <= range[1]; ++i)
+				expect(arrayHelpers.alignUp(i, alignment)).to.equal(expectedValue);
+		};
+
+		it('always aligns up', () => {
+			assertAlignUp([0, 0], 8, 0);
+			assertAlignUp([1, 8], 8, 8);
+			assertAlignUp([9, 16], 8, 16);
+			assertAlignUp([257, 264], 8, 264);
+		});
+
+		it('can align using custom alignment', () => {
+			assertAlignUp([0, 0], 11, 0);
+			assertAlignUp([1, 11], 11, 11);
+			assertAlignUp([12, 22], 11, 22);
+			assertAlignUp([353, 363], 11, 363);
+		});
+	});
+
+	class FakeFactory {
+		constructor(sizes) {
+			this.sizes = sizes;
+			this.index = 0;
+		}
+
+		deserialize(buffer) {
+			return { size: this.sizes[this.index++], tag: buffer.byteOffset };
+		}
+	}
+
+	class TestContext {
+		constructor(sizes, viewSize = 52) {
+			this.byteArray = new Uint8Array(100);
+			this.subView = new Uint8Array(this.byteArray.buffer, 15, viewSize);
+			this.factory = new FakeFactory(sizes);
+		}
+	}
+
+	const assertThrowsWhenAnyElementHasZeroSize = reader =>
```

Comment:
reader => view ?

Reply:
this time it's actually reader not a bug